### PR TITLE
Event check-in backend

### PR DESCRIPTION
**Pull request by alicesf2**
Refs #281

### CODE REVIEW

File: backend/src/api/events.js [hunk 1]
```diff
@@ -99,6 +100,66 @@ router.put(
...
+
+      await event.save()
+      await attendee.save()
+
+      res.json({
+        code: 200,
+        message: 'Successfully Checked In',
+        success: true
+      })
+    } else {
```
Comment (alanfang7) [inline backend/src/api/events.js]: might be more readable if you did:

```
if (!event) {
    return 404
}
else {
// do whatever
}
```

but it's not really a huge deal

File: backend/src/utils/utils.js [hunk 2]
```diff
@@ -11,6 +12,17 @@ const getStats = async () => {
   return newStats
 }
 
+const attendeeIsLate = event => {
+  const curMoment = moment()
+  const startMoment = moment(`${event.date} ${event.startTime}`)
+
+  if (curMoment.isAfter(startMoment)) {
```
Comment (alanfang7) [inline backend/src/utils/utils.js]: perhaps just do `return curMoment.isAfter(startMoment)`

File: backend/src/models/event.js [hunk 3]
```diff
@@ -8,8 +7,8 @@ const Event = new mongoose.Schema({
   endTime: { type: String, required: true },
   location: { type: String, required: true },
   description: { type: String, required: true },
-  attendees: { type: [mongoose.Schema.Types.ObjectId], default: [] },
-  lateAttendees: { type: [mongoose.Schema.Types.ObjectId], default: [] },
+  attendees: { type: [String], default: [] },
```
Comment (arpanlaha) [inline backend/src/models/event.js]: Does it just track by name then?

Comment (alanfang7) [reply backend/src/models/event.js]: maybe store this as `attendeeEmails` and `lateAttendeeEmails`

Comment (alicesf2) [reply backend/src/models/event.js]: yea it tracks by email

File: backend/src/api/events.js [hunk 4]
```diff
@@ -99,6 +100,66 @@ router.put(
...
+    const eventId = req.params.eventId
+    const attendeeEmail = data.email
+
+    const event = await Event.findById(eventId)
+    if (event) {
+      let attendee = await Attendee.findOne({ email: attendeeEmail })
+
+      // create new attendee if not in db
+      if (!attendee) {
+        const newAttendee = new Attendee({
```
[the remark this answers is not in the document]
Comment (alicesf2) [reply backend/src/api/events.js]: ok

File: backend/src/models/event.js (hunk 3 above)
Comment (alicesf2) [reply backend/src/models/event.js]: ok

File: backend/src/utils/utils.js (hunk 2 above)
Comment (alicesf2) [reply backend/src/utils/utils.js]: ok

File: backend/src/api/events.js (hunk 1 above)
Comment (alicesf2) [reply backend/src/api/events.js]: ok

File: backend/src/api/events.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (alicesf2) [reply backend/src/api/events.js]: why did u use ur work email smh

**Pull request merged**